packages/gatsby-link/src/index.js

```diff
@@ -8,7 +8,11 @@ import { parsePath } from "./parse-path"
 export { parsePath }
 
 export function withPrefix(path) {
-  return normalizePath(`${__PATH_PREFIX__}/${path}`)
+  return normalizePath(`${__BASE_PATH__}/${path}`)
+}
+
+export function withAssetPrefix(path) {
+  return `${__PATH_PREFIX__}/${path}`
```

Comment:
why is the `withAssetPrefix` not populated with `__ASSET_PREFIX__`

from:
https://github.com/gatsbyjs/gatsby/blob/39403b8ac211d9c70b7fc028772aed37a1970807/packages/gatsby/src/utils/webpack.config.js#L184-L186

Reply:
is here no `normalizePath` necessary?

Reply:
__PATH_PREFIX__ itself is populated with assetPrefix. normalizePath would strip the double slashes, e.g. if assetPrefix is set to `https://cdn.example.org` and `pathPrefix` is set to `/blog`, normalizePath would create a structure like `https:/cdn.example.org/blog/${path}` which is not what we want.

Reply:
And it _is_ prefixed with __ASSET_PREFIX__. We want to use the full path here, e.g. assetPrefix joined with pathPrefix.